examples/dlp/cloud_function_example/cloud_function/main.py

```diff
@@ -0,0 +1,154 @@
+  """
+
+  pubsub_message = base64.b64decode(event['data']).decode('utf-8')
+  json_payload = json.loads(pubsub_message)
+  return json_payload
+
+
+def _build_deidentify_config(info_type_list):
+  """Constructs the deidentify config dictionary for the dlp api.
+
+    Args:
+      info_type_list: A list containing the dlp info types to redact.
+
+    Returns:
+      A dictionary representation of the DLP API DeidentifyConfig object
+      to be used to redact the sensitive fields.
+  """
+
+  transformation_config = {
+      'info_types': [],
+      'primitive_transformation': {
+          'replace_config': {
+              'new_value': {
+                  'string_value': ''
+              }
+          }
+      }
+  }
+
+  deidentify_config = {'info_type_transformations': {'transformations': []}}
+
+  for info_type in info_type_list:
+    config = copy.deepcopy(transformation_config)
+    config['info_types'].append({'name': info_type})
+    replace_config = config['primitive_transformation']['replace_config']
+    replace_config['new_value']['string_value'] = '[%s]' % info_type
+    deidentify_config['info_type_transformations']['transformations'].append(
+        config)
+  return deidentify_config
+
+
+def _get_redacted_dlp_response(message):
+  """Calls the DLP API and returns the response.
+
+    Args:
+      message: The string that should be inspected by the DLP API.
+
+    Returns:
+      A DLP API deidentify content response.
+  """
+
+  dlp = dlp_v2.DlpServiceClient()
+  info_types = [{'name': info_type} for info_type in _INFO_TYPE_LIST]
+  inspect_config = {'info_types': info_types}
+  deidentify_config = _build_deidentify_config(_INFO_TYPE_LIST)
+  item = {'value': message}
+  parent = dlp.project_path(_PROJECT_ID)
+  return dlp.deidentify_content(
+      parent,
+      inspect_config=inspect_config,
+      deidentify_config=deidentify_config,
+      item=item)
+
+
+def _post_scrubbed_message_to_pub_sub(message_topic, scrubbed_message,
+                                      log_json):
+  """Posts the log json payload with redacted PII to a pub/sub topic.
+
+    Args:
+      message_topic: The pub/sub topic to post the message to.
+      scrubbed_message: The PII scrubbed message string.
+      log_json: The original log json.
+  """
+
+  publisher = pubsub_v1.PublisherClient()
+  topic_path = publisher.topic_path(_PROJECT_ID, message_topic)
+  log_json['jsonPayload']['message'] = scrubbed_message
+  data = json.dumps(log_json).encode('utf-8')
+  message_future = publisher.publish(topic_path, data=data)
+  message_future.add_done_callback(_post_scrubbed_message_callback)
+
+
+def _post_scrubbed_message_callback(message_future):
+  """The pub/sub message post callback.
+
+    Args:
+      message_future: A google.api_core.future.Future object returned from the
+        pub/sub publish call.
+  """
+
+  if message_future.exception(timeout=30):
+    print('Publishing message on {} threw an Exception {}.'.format(
```

Comment:
I know that CFs will log stdout but refactor to use `logging.info` for clarity.

Reply:
Done